### PR TITLE
Add webp support for cards images

### LINKED ISSUE #643: Image support for webp
Webp images are smaller by same quality, so it would be nice if the GUI would support to use webp images too

### PR DESCRIPTION
**Pull request by add-le**
Add support webp format to the cards images :
Give 25% to 50% smaller image size for same quality as jpg.
Can reduce the cache folder to 25% to 50% its size, for android support can be a good idea.

https://developers.google.com/speed/webp
https://developers.google.com/speed/webp/docs/webp_study

If you find the good settings to convert all superLQ's kevlahnota to webp, without losing much quality, you can achieve near to 30-35% reduce size.

- [ ] Need to check if it's work for token and crop images too

Resolves #643

### CODE REVIEW

**Comment by kevlahnota** [comment]
Change title to add webp support for desktop version, mobile version uses libgdx and it doesn't support webp. There's a webp textureloader here: https://github.com/ZumiKua/libgdx-webp-loader
But you need modify how it load images.

**Review by add-le — COMMENTED** [review]
@kevlahnota [e37f652](https://github.com/Card-Forge/forge/pull/1802/commits/e37f652e4f761492ef1c36389325a052dd41ef83) Is it the right way to add the lib loader ? I don't know too much about maven and this gdx system. I don't think I know how it works.

**Comment by add-le** [comment]
Still on it

File: forge-core/src/main/java/forge/ImageKeys.java [hunk 1]
```diff
@@ -66,7 +66,7 @@ public static void initializeDirs(String cards, Map<String, String> cardsSub, St
 
     // image file extensions for various formats in order of likelihood
     // the last, empty, string is for keys that come in with an extension already in place
-    private static final String[] FILE_EXTENSIONS = { ".jpg", ".png", "" };
+    private static final String[] FILE_EXTENSIONS = { ".jpg", ".png", "webp", "" };
```
Comment (myyk) [inline forge-core/src/main/java/forge/ImageKeys.java:69]: Did you mean `.webp`?

```suggestion
    private static final String[] FILE_EXTENSIONS = { ".jpg", ".png", ".webp", "" };
```

**Comment by CTimmerman** [comment]
`Forge\Cache\pics\cards\DOM\Jodah, Archmage Eternal.fullborder.jpg` is only 86,4 KiB (88.552 bytes) on my Windows 11. Paint.Net says WebP at 83% quality is about the same size at 86,8 KiB (88.920 bytes), but both take 88,0 KiB (90.112 bytes) in NTFS. Q10 AVIF (unsupported on GitHub) and Q50 WEBP (top) look similar to the original JPG (bottom) though, at 48 KiB on disk (55%, so saving 45% space, which is significant as my current Cache folder is 1,41 GiB (1.523.142.656 bytes) on disk):
Jodah, Archmage Eternal fullborder_q50
Jodah, Archmage Eternal fullborder
Not sure how much GitHub's conversion affects those, so here's a .zip:
[test_mtg_file_size.zip](https://github.com/user-attachments/files/26512927/test_mtg_file_size.zip)

**Pull request closed without merging**